src/Entity/ApiDoc.php

```diff
@@ -231,7 +278,35 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setLabel(t('Changed'))
       ->setDescription(t('The time that the entity was last edited.'));
 
+    $fields['fetched_timestamp'] = BaseFieldDefinition::create('timestamp')
+      ->setLabel(t('Spec fetched from URL timestamp'))
+      ->setDescription(t('When the OpenAPI spec file was last fetched from URL as a Unix timestamp.'));
+
     return $fields;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function preSave(EntityStorageInterface $storage) {
+    parent::preSave($storage);
+
+    \Drupal::service('apigee_api_catalog.spec_fetcher')->fetchSpec($this, FALSE, FALSE);
```

Comment:
https://github.com/apigee/apigee-edge-drupal/pull/191/files#r283731935

![image](https://user-images.githubusercontent.com/1755573/58000786-1547d880-7ada-11e9-879e-a003e9852152.png)

Reply:
I prefer to keep code/classes to a minimum whenever possible and creating a new class and overriding a storage handler just to use dependency injection design pattern seems unnecessary to me.

Reply:
Handling things in the right place it is never an unnecessary work :) It is not just about DI, when Drupal core maintainers finally deprecate and remove CRUD methods from entity classes then leaving this here will cause extra work for us that we could have eliminated before this PR got merged.
If we'll need to create a storage class for this entity type later, what ensures this code will be moved to its rightful place at that time? Nothing.

Reply:
Drupal core uses this static service resolver ~825 times in runtime code. ~200 times in core libraries (as opposed to modules) so It's not a terrible thing.

Reply:
Well, just because you see a pattern repeating in Drupal 8 core it does necessarily mean that is a good architectural solution. This is the reason why I learning from other systems and reading books about software-architecture and OOP coding to extend my current knowledge and do not just blindly trust what I see in Drupal core.
Let's take `final` classes as an example, you do not see in Drupal core and probably that is one of the reasons why BC breaking is so easy and sometimes upgrade path suffers from changes made in Drupal core on non-final classes or properties and methods that were public/protected.
Also do not forget the OOP was a huge switch compared with Drupal 7 for everyone, even for some Drupal maintainer. So the Drupal 8 core currently contains initial decisions that they made in the beginning and now they are trying to get rid of that by using deprecations. The problem is with deprecations that it does not allow to change everything that they want.